utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,83 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
-                None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
-                }
-            }
-        };
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes(&self, sys_calls: &[SysCallName]) -> Option<Vec<usize>> {
```

Comment:
Basically, I guess the `invocable_sys_calls_indexes` will be called from a method, which builds precise sys-calls invocation. That means such method will "know" sys-calls it needs. So `invocable_sys_calls_indexes` can accept not the slice, but the `[SysCalls; N]` and give as a result array of the same size. This allows us to avoid additional vector allocation